server/src/lib/migrate/mod.rs

```diff
@@ -331,10 +331,12 @@ pub async fn dest_initiate(
     // We've successfully negotiated a migration protocol w/ the source.
     // Now, we spawn a new task to handle the actual migration over the upgraded socket
     let mctx = migrate_context.clone();
-    let task_rqctx = rqctx.clone();
+    let context = rqctx.context().context.lock().await;
+    let context = context.as_ref().unwrap();
```

Comment:
You can avoid this `unwrap` by just matching what the source side code does.

Reply:
Done.

(Though it's not clear how we can get to this point without context being initialized. Anyway, defense in depth!)